sys/arm64/arm64/elf_machdep.c

```diff
@@ -763,13 +767,20 @@ elf_reloc_self(const Elf_Dyn *dynp, void *data_cap, const void *code_cap)
 
 	for (; rela < rela_end; rela++) {
 		/* Can not panic yet */
-		if (ELF_R_TYPE(rela->r_info) != R_MORELLO_RELATIVE)
-			continue;
-
-		fragment = (Elf_Addr *)((char *)data_cap + rela->r_offset);
-		cap = build_cap_from_fragment(fragment, 0, rela->r_addend,
-		    data_cap, code_cap);
-		*((uintptr_t *)fragment) = cap;
+		switch (ELF_R_TYPE(rela->r_info)) {
+		case R_AARCH64_RELATIVE:
+		case R_AARCH64_FUNC_RELATIVE:
+			/* XXX Dapeng: Which base to use here? */
+			break;
```

Comment:
As discussed on Zoom: the kernel is position-dependent so should not have any R_AARCH64_RELATIVE relocations, but we should check that LLD definitely won't emit them for a ptraddr_t global in a purecap kernel. If it does emit them for whatever reason, relocbase is 0 for the kernel (due to being linked at an absolute address), and that is what the 0 in the call to build_cap_from_fragment is for, so these relocations would be a no-op anyway. The only reason R_MORELLO_RELATIVE is needed (and, soon, FUNC_RELATIVE) is because of the need to derive valid caps at run time for their tags.